### DISCUSSION TITLE
Transfer gpuarray to tensor in gpu

### DISCUSSION THREAD

**Discussion by KaiTang98** [inducer/pycuda#412] [Q&A]
Hi, I try to use pycuda to create pinned memory for spped up the transfer speed from cpu to gpu. I can btain a gpuarray in gpu after doing the 

cuda.memcpy_htod(devPtr, img_pin)
gpu_array = gpuarray.GPUArray(test_img_trasform.shape, np.float32, gpudata=devPtr)

How can I transfer this gpu_array to tensor in gpu?

Thanks in adcanced.

**Comment by inducer** [comment]
Does this example help?

https://github.com/inducer/pycuda/blob/main/examples/cai_cupy_arrays.py
